### PR TITLE
example(Widgets): add an example of widgets

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Add an example of widgets.

### CODE REVIEW

File: examples/js/GUI/Widgets.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    animationDuration: 500,
+    position: 'bottom-left',
+    direction: 'column',
+}
+
+
+class Widgets {
+    constructor(view, options = {}) {
+        // ---------- BUILD PROPERTIES ACCORDING TO DEFAULT OPTIONS AND OPTIONS PASSED IN PARAMETERS : ----------
+
```
Comment (gchoqueux) [inline examples/js/GUI/Widgets.js]: could you add translate options.

**Review by gchoqueux — COMMENTED** [review]
could you add doc?
could use the bundle dist with my branch ?
maybe rename the bundle `itowns_widgets`
Could you rename `Widgets` class to `navigation`? check on other API

File: examples/misc_widgets.html [hunk 2]
```diff
@@ -61,7 +36,7 @@
             const placement = {
                 coord: new itowns.Coordinates('EPSG:4326', 2.351323, 48.856712),
                 range: 6000,
-                heading: 45,
+                // heading: 45,
```
Comment (gchoqueux) [inline examples/misc_widgets.html]: could you remove comment?

File: utils/gui/Widgets.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    display3DToggle: true,
+    displayZoomIn: true,
+    displayZoomOut: true,
+    animationDuration: 500,
+    position: 'bottom-left',
+    direction: 'column',
+};
+
+
+class Widgets {
```
Comment (gchoqueux) [inline utils/gui/Widgets.js]: could you add documentation?

File: utils/gui/Widgets.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        const buttonBar = document.createElement('div');
+        buttonBar.className = 'widget-button-bar';
+        if (id) { buttonBar.id = id; }
+
+        this.domElement.appendChild(buttonBar);
+        return buttonBar;
+    }
+}
+
+export default Widgets;
```
Comment (gchoqueux) [inline utils/gui/Widgets.js]: could you rename `Widgets` to `navigation`

File: examples/misc_widgets.html [hunk 5]
```diff
@@ -101,22 +76,30 @@
...
+                'button-example-2',
+                '2',
+                () => { alert('button 2 clicked'); },
+                'button-bar-example-1',
+            );
+            widgets.addButton(
+                'button-example-3',
+                '3',
+                () => { alert('button 3 clicked'); },
+            );
```
Comment (gchoqueux) [inline examples/misc_widgets.html]: could you add real user case with buttons?

File: webpack.config.js [hunk 6]
```diff
@@ -58,6 +58,10 @@ module.exports = () => {
                 import: './utils/debug/Main.js',
                 dependOn: 'itowns',
             },
+            itowns_gui: {
```
Comment (gchoqueux) [inline webpack.config.js]: could you rename to `itowns_widget`?

File: examples/misc_widgets.html (hunk 2 above)
Comment (mgermerie) [reply examples/misc_widgets.html]: Removed.

File: utils/gui/Widgets.js (hunk 4 above)
Comment (mgermerie) [reply utils/gui/Widgets.js]: I renamed it.

File: utils/gui/Widgets.js (hunk 3 above)
Comment (mgermerie) [reply utils/gui/Widgets.js]: I added doc for the plugin

File: webpack.config.js (hunk 6 above)
Comment (mgermerie) [reply webpack.config.js]: renamed.

File: examples/misc_widgets.html (hunk 5 above)
Comment (mgermerie) [reply examples/misc_widgets.html]: I changed the example buttons.

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**